plugins/utils/ollama.nix

```diff
@@ -0,0 +1,231 @@
+                -- when a response is received.
+                ---@type Ollama.PromptActionResponseCallback
+                return function(body, job)
+                  -- body is a table of the json response
+                  -- body.response is the response text received
+
+                  -- job is the plenary.job object when opts.stream = true
+                  -- job is nil otherwise
+                end
+
+              end
+            '';
+          };
+
+          opts = {
+            stream = helpers.defaultNullOpts.mkBool false ''
+              Whether to stream the response.
+            '';
+          };
+        };
+      })
+    ];
+in {
+  meta.maintainers = [maintainers.GaetanLepage];
+
+  options.plugins.ollama =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "ollama.nvim";
+
+      package = helpers.mkPackageOption "ollama.nvim" pkgs.vimPlugins.ollama-nvim;
+
+      model = helpers.defaultNullOpts.mkStr "mistral" ''
+        The default model to use.
+      '';
+
+      prompts = let
+        promptOptions = {
+          prompt = mkOption {
+            type = types.str;
+            description = ''
+              The prompt to send to the model.
+
+              Replaces the following tokens:
+              - `$input`: The input from the user
+              - `$sel`: The currently selected text
+              - `$ftype`: The filetype of the current buffer
+              - `$fname`: The filename of the current buffer
+              - `$buf`: The contents of the current buffer
+              - `$line`: The current line in the buffer
+              - `$lnum`: The current line number in the buffer
+            '';
+          };
+
+          inputLabel = helpers.defaultNullOpts.mkStr "> " ''
+            The label to use for an input field.
+          '';
+
+          action = helpers.mkNullOrOption actionOptionType ''
+            How to handle the output.
+
+            See [here](https://github.com/nomnivore/ollama.nvim/tree/main#actions) for more details.
+
+            Defaults to the value of `plugins.ollama.action`.
+          '';
+
+          model = helpers.mkNullOrOption types.str ''
+            The model to use for this prompt.
+
+            Defaults to the value of `plugins.ollama.model`.
+          '';
+
+          extract =
+            helpers.defaultNullOpts.mkNullable
+            (with types; either str (enum [false]))
+            "```$ftype\n(.-)```"
+            ''
+              A `string.match` pattern to use for an Action to extract the output from the response
+              (Insert/Replace).
+            '';
+
+          options = helpers.mkNullOrOption (with types; attrsOf anything) ''
+            Additional model parameters, such as temperature, listed in the documentation for the [Modelfile](https://github.com/jmorganca/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values).
+          '';
+
+          system = helpers.mkNullOrOption types.str ''
```

Comment:
Please allow a rawType with `helpers.maybeRaw`

Reply:
`helpers.maybeRaw` is not a thing. It is not exposed by the helpers but only used internally in `defaultNullOpts.*`.

Reply:
Then we shoule either expose the maybeRaw, or put an either here

Reply:
Done (in a separate commit)